### PR TITLE
Helper

### LINKED ISSUE #12: Matrix -- Move components out of Helper package into Matrix package, with some tweaks
- [x] Rename function to a verb. Like `generateFibMatrix`
- [x] Move error handling into generateFibMatrix function (After changes above, the function will output both the matrix AND error message)
- [x] Update the testing to reflect the changes to the function

### PR DESCRIPTION
**Pull request by sarahhtbarton**

### CODE REVIEW

File: internal/matrix/matrix.go
Comment (AYM1607) [inline internal/matrix/matrix.go]: delete extra spaces

File: internal/server.go
Comment (AYM1607) [inline internal/server.go]: This is definitely going to cause a merge conflict. Your other PR moves this logic to a different package so I'd advise to merge that one into master first, then merge master into this branch and fix the inevitable conflicts

File: internal/matrix/matrix_test.go
Comment (AYM1607) [inline internal/matrix/matrix_test.go]: are you running a go formatter with your IDE? this doesn't seem to be following formatting

File: internal/matrix/matrix_test.go
Comment (AYM1607) [inline internal/matrix/matrix_test.go]: As advised in a different PR, try to use https://pkg.go.dev/github.com/stretchr/testify/require for test expectations

**Review by AYM1607 — CHANGES_REQUESTED** [review]
I'd wait until you merge #18 and you merge master into this branch before doing more changes

File: internal/matrix/matrix.go
Comment (AYM1607) [inline internal/matrix/matrix.go:17]: It might be a good idea to declare these errors as variables, e.g. `dimensionError` and `dimensionOutOfBoundsError` just to say something. The problem here is that if you ever change the wording of your error message, your tests will break unnecessarily. There's more advance techniques you could use here like declaring a custom error type and testing against type instead of value, but that's a bit to overkill for this use case. You could also use simpler techniques such as only testing for error existence and not equality to avoid the false positive in the future. I'll leave the decision to you. We can talk about this in more detail if you'd  like.

**Review by AYM1607 — CHANGES_REQUESTED** [review]

File: internal/matrix/matrix.go
Comment (sarahhtbarton) [reply internal/matrix/matrix.go:17]: Thats a great callout. I did change the wording and did unnecessarily break my tests so have already encountered this problem and would like to avoid it again in the future! Will look into the different approaches now

Comment (sarahhtbarton) [reply internal/matrix/matrix.go:17]: I went with checking for the existence of an error, but I'd be interested in talking through how declaring errors as variables would work. Ill add it to our 1x1 agenda for tomorrow

File: internal/matrix/matrix_test.go
Comment (AYM1607) [inline internal/matrix/matrix_test.go:17]: Changing this to be type bool should make it more readable, but if tests pass like this then it should be fine.

**Review by AYM1607 — APPROVED** [review]
LGTM, left a little suggestion.

**Pull request merged**
